scripts/Muon/GUI/Common/fitting_widgets/basic_fitting/basic_fitting_model.py

```diff
@@ -491,7 +491,8 @@ def _get_new_end_x_for(self, new_dataset_name: str) -> float:
         if new_dataset_name in self.dataset_names:
             return self.end_xs[self.dataset_names.index(new_dataset_name)]
         else:
-            return self.current_end_x
+            x_lower, x_upper = self.x_limits_of_workspace(new_dataset_name)
+            return self.current_end_x if x_lower < self.current_end_x < x_upper else x_upper
```

Comment:
I dont think we can do 2 comparisons in this line ` x_lower < self.current_end_x < x_upper` unless I have missed something

Reply:
I think this should be fine, I have tested it and it doesn't complain. When I do the comparisons like `self.current_end_x > x_lower and self.current_end_x < x_upper` pycharm tells me to write it the above way

Reply:
huh, must be a newish thing